src/main/resources/org/jpeek/metrics/TLCOM.xsl

```diff
@@ -0,0 +1,147 @@
+<?xml version="1.0"?>
+<!--
+The MIT License (MIT)
+
+Copyright (c) 2017-2018 Yegor Bugayenko
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0">
+  <xsl:template match="skeleton">
+    <metric>
+      <xsl:apply-templates select="@*"/>
+      <title>TLCOM</title>
+      <description>
+        <xsl:text>
+          "Transitive Lack of Cohesion in Methods" (TLCOM) expands upon
+          the original definition for LCOM by extending the criteria for
+          estabilishing relations between method pairs by also including cases
+          when a m1 calls m2 and m2 happens to use attribute a2 of the class. This
+          then means that m1 transitively uses a2 via m2.
+          The chain of calls can have more than one hop: if m1 calls m2 which calls
+          m3 which calls m4, and m4 uses attribute a4, then m1 is said to transitively
+          use a4 as well.
+          The final formula is the same as the original LCOM:
+          TLCOM = |P| - |Q|, if |P| &gt; |Q|, otherwise it's 0 (zero). 'P' is
+          the set of method-pairs that are not connected according to the
+          definition above, and 'Q' is the set of method-pairs that are connected.
+        </xsl:text>
+      </description>
+      <xsl:apply-templates select="node()"/>
+    </metric>
+  </xsl:template>
+  <xsl:template match="class">
+    <!--
+    @todo #65:30min TLCOM: we are including access to ALL fields here. After #156 is fixed,
+     come back and refactor xpath for $Ms_that_use_attrs, $left_attrs, and $right_attrs
+     in order to avoid including access to fields belonging to other classes.
```

Comment:
@llorllale what are "other classes"? Extended classes, inner classes, or?

Reply:
@amihaiemil classes other than the one being analyzed